### PR TITLE
Require distinct image for `kubectl rolling-update NAME --image=IMAGE`

### LINKED ISSUE #9045: kubectl rolling-update should not exitcode=0 when resumes an update
Consider this:

```
> kubectl rolling-update A --image a:v2
```
Control-C half way through update

```
> kubectl rolling-update A --image a:v3
Found existing update in progress (X), resuming.
Continuing update with existing controller (X).
Update succeeded. Deleting old controller: X
Renaming X to X
```

So the update to v3 succeeded (I think), but the update was actually to v2.

I think it should update to v3; ideally by co-opting the existing rolling-update and repointing to v3.  If we can't do that, it should completing the existing rolling-update, then rolling-update to v3.  If we can't have that, it should exit-code != 0.

### PR DESCRIPTION
**Pull request by j3ffml**
Fixes #9045

@kubernetes/kubectl

### CODE REVIEW

**Comment by jlowdermilk** [comment]
In case it's not clear why we want this restriction, it's to try to prevent users getting themselves in a weird state, and in particular to block use of this command with the `image:latest` tag. A rolling-update of all rc-managed pods to the same image is not guaranteed to have the desired effect, because image caching on the node may result in the new pods using the old "version" of the image.

The assumption that `--image` specified a distinct image was always there, this just enforces it. If you want to do a rolling-update with the same image:tag, you can use `kubectl rolling-update -f rc-definition.yaml`.

File: pkg/kubectl/cmd/rollingupdate.go [hunk 1]
```diff
@@ -229,6 +229,12 @@ func RunRollingUpdate(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, arg
 	// than the old rc. This selector is the hash of the rc, which will differ because the new rc has a
 	// different image.
 	if len(image) != 0 {
+		if len(oldRc.Spec.Template.Spec.Containers) > 1 {
+			return cmdutil.UsageError(cmd, "Image update is not supported for multi-container pods")
```
Comment (bgrant0607) [inline pkg/kubectl/cmd/rollingupdate.go]: Don't we already have an error for that somewhere? #13483

**Comment by bgrant0607** [comment]
Cases to think about:
1. Resuming an interrupted rolling update
2. Rollback

As long as you're doing this, could you also address #9045?

File: pkg/kubectl/cmd/rollingupdate.go (hunk 1 above)
Comment (jlowdermilk) [inline pkg/kubectl/cmd/rollingupdate.go]: Yes, in rolling_updater. Removed superfluous check.

**Comment by jlowdermilk** [comment]
PTAL. I changed it to print the error message about requiring a distinct image name when starting a fresh rolling update. Added check for matching image name on resuming an interrupted rolling update that tells the user to either continue the existing rollback with the in-progress image, or revert it with `--rollback`.

**Comment by bgrant0607** [comment]
Thanks much. LGTM.

**Pull request merged**